src/service_account.rs

```diff
@@ -192,11 +193,23 @@ impl<'a, C> ServiceAccountAccess<C>
             client: client,
             key: key,
             cache: MemoryStorage::default(),
+            sub: None,
+        }
+    }
+
+    pub fn with_sub(key: ServiceAccountKey, client: C, sub: Option<String>) -> ServiceAccountAccess<C> {
```

Comment:
how about making the `sub` argument a non-Option `String`? After all, if no sub is needed, `new()` should be used anyway.